Inc/DirectXMath.h

```diff
@@ -151,17 +151,25 @@
 #if defined(_XM_SSE_INTRINSICS_) && !defined(_XM_NO_INTRINSICS_)
 
 #if defined(_XM_NO_MOVNT_)
-#define XM_STREAM_PS( p, a ) _mm_store_ps( p, a )
+#define XM_STREAM_PS( p, a ) _mm_store_ps((p), (a))
 #define XM_SFENCE()
 #else
-#define XM_STREAM_PS( p, a ) _mm_stream_ps( p, a )
+#define XM_STREAM_PS( p, a ) _mm_stream_ps((p), (a))
 #define XM_SFENCE() _mm_sfence()
 #endif
 
+#if defined(_XM_FMA3_INTRINSICS_)
+#define XM_FMADD_PS( a, b, c ) _mm_fmadd_ps((a), (b), (c))
```

Comment:
Introducing this macro avoids having a bunch of new paths. Also makes it easier to experiment with using FMA3 vs. not.